compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -37,13 +37,17 @@ pub fn is_const_evaluatable<'tcx>(
         | ty::ConstKind::Value(_, _)
         | ty::ConstKind::Error(_) => return Ok(()),
         ty::ConstKind::Infer(_) => return Err(NotConstEvaluatable::MentionsInfer),
-    };
+    }
 
     if tcx.features().generic_const_exprs {
         let ct = tcx.expand_abstract_consts(unexpanded_ct);
 
+        if trivially_satisfied_from_param_env(ct, param_env) {
```

Comment:
This ought to go *before* the `expand_abstract_consts` call as otherwise its never going to find any matches in the env since abstract consts arent expanded in the `ParamEnv` :3